src/Indice.Features.Cases.AspNetCore/Controllers/AdminCasesController.cs

```diff
@@ -247,8 +251,24 @@ internal class AdminCasesController : ControllerBase
         }
 
         private async Task<byte[]> CreatePdf(Case @case) {
+            var isPortrait = true;
+            var requiresQrCode = false;
+            if (@case.CaseType.Config is not null) {
+                var caseTypeConfig = JsonSerializer.Deserialize<JsonDocument>(@case.CaseType.Config);
+                if (caseTypeConfig.RootElement.TryGetProperty("IsPortrait", out var isPortraitConfig)) {
+                    isPortrait = isPortraitConfig.GetBoolean();
+                }
+                if (caseTypeConfig.RootElement.TryGetProperty("RequiresQrCode", out var requiresQrCodeConfig)) {
+                    requiresQrCode = requiresQrCodeConfig.GetBoolean();
+                }
+            }
             var template = await _caseTemplateService.RenderTemplateAsync(@case);
-            return await _casePdfService.HtmlToPdfAsync(template);
+            var byteArray = await _casePdfService.HtmlToPdfAsync(template, isPortrait);
+            if (requiresQrCode) { // some case pdfs need to have a QR Code, but others don't
+                // add QR Code
+                byteArray = _qrCodeService.Add(byteArray, @case.Id);
```

Comment:
service που μπορεί να πάρει το bytearray ενος pdf και να βάλει qrcode δεν μου φαίνεται ανεξάρτητο. Είναι τελείως δεμένο με τον pdf reader και υποθέτω και το formatting του pdf αυτού καθεαυτού. συνεπώς θα μου φαινόταν λογικότερο στο pdfService, σαν εξτρα φλαγκ στο html2pdf

Reply:
έτσι ήταν πριν και είχε καταλήξει το signature HtmlToPdfAsync(template, isPortrait, requiresQrCode, @case.Id, digitallySigned)
Για αυτό αποφάσισα να τα σπάσω σε μικρότερα μέρη.

Reply:
Το προβλημα είναι πως αμα μπει κεντρικά, όλα τα pdf θα έχουν κοινό config, άσχετα αν αφορουν admin ή my ? Ή δεν είναι προβλημα, γιατί πάντα είναι 1 το config?

Παντως, αν ειναι να ξαναμπει εκεί που ήταν μπορει να γινει και σαν extension του αρχικου interface ωστε να μην χαλάμε το contract κάθε φορά, ή με pdf options ώστε να μπουν ευκολα και μελλοντικά options

Reply:
The signature now is `HtmlToPdfAsync(string htmlTemplate, PdfOptions pdfOptions, Case @case)`